Clear filters when selecting curated collections

nicotaglienti opened the pull request:
This PR fixes #431 , Filters must be cleared when users click on a card of the Curated Collection section.-

**To Test**

- On the dev branch
- Apply some filters and then click on one card from the Curated Collection section
- Notice the previous filters are still applied
- Checkout this branch and make sure the filters are cleared before applying the card's filter.-
- Also notice you have clearFiltersForCuratedCollections.spec that covers the most common uses of this feature

fdalcin reviewed (changes requested):
Hi @nicotaglienti, thanks for your contribution.

I noticed just a couple of things:

1. After applying a search query and then clicking on one of the Curated Collections the URL resets but the search input field keeps the old value
2. Are you willing to try to identify why tests are failing on the pipeline?

One thing that's not related to the PR, but could remove `.only` from this test `Removing chip will remove the date filter`? This is preventing Cypress "run all specs" to run properly.

nicotaglienti commented:
yes sure! I'll take a look and try to fix it!

nicotaglienti commented:
by the way, the "Clear all" link doesn't clear the query search input either, should it do it?

fdalcin commented:
Hi @nicotaglienti. It's ok that "Clear all" does not clear the query since it should only remove applied filters, while clicking on a Curated Collection is more like a full query reset.

nicotaglienti commented:
> Hi @nicotaglienti, thanks for your contribution.
> 
> I noticed just a couple of things:
> 
> 1. After applying a search query and then clicking on one of the Curated Collections the URL resets but the search input field keeps the old value
> 2. Are you willing to try to identify why tests are failing on the pipeline?
> 
> One thing that's not related to the PR, but could remove `.only` from this test `Removing chip will remove the date filter`? This is preventing Cypress "run all specs" to run properly.

Hi @fdalcin,

In order to finish this PR, I fixed what you mentioned in point one, and also added one more test to verify this. On the other hand, I could not find a solution for the test failure, it seems that it does not refer to the code but to the environment, you can see here:

https://github.com/pressbooks/pressbooks-book-directory-fe/runs/4382224743?check_suite_focus=true

CY it's not finding the elements of the page just like if the site it's not live/build, and tests are running ok on my local machine, so I don't know how to help you.

Sorry if I'm wrong since it's my first time working with cypress.

Please let me know if I can make something to help you with this problem.-

All Best!

fdalcin commented:
Hi @nicotaglienti,

Thanks, I'll take a look at it tomorrow. 
Tests are working fine locally for me too, I'll try to find what's causing issues to the pipeline as well.

fdalcin commented:
Hi @nicotaglienti 

I was checking the issue with GitHub Actions yesterday. I created a new test branch with your changes and tests work properly, this might be related to some cache issue or something.

Are you willing to close this and open a new PR? Hopefully the issue will go away.

nicolasresetdigital commented:
> Hi @nicotaglienti
> 
> I was checking the issue with GitHub Actions yesterday. I created a new test branch with your changes and tests work properly, this might be related to some cache issue or something.
> 
> Are you willing to close this and open a new PR? Hopefully the issue will go away.

Sure! I'll do it today
